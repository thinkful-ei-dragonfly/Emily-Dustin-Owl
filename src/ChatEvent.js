import React from 'react';
import STORE from './Store';


export default function(props) {
  if(props.type === 'message') {
    return (
      <div className='message'>
        <img src={STORE.participants.find(obj => obj.id === props.participantId).avatar} alt='user avatar'></img>
        <h3>{STORE.participants.find(obj => obj.id === props.participantId).name}</h3>
        <p>{new Date(props.time).toLocaleTimeString()}</p>
        <p className = 'message-text'>{props.message}</p>
      </div>
    )
  }
  else if(props.type === 'thumbs-up'){
    return (
      <div className ='thumbs-up'>
        <p><strong>{STORE.participants.find(obj => obj.id === props.participantId).name}</strong>gave a thumbs up 
        <i className="fas fa-thumbs-up"></i></p>
      </div>)

  }
  else if (props.type === 'thumbs-down'){
    return (
      <div className ='thumbs-down'>
        <p><strong>{STORE.participants.find(obj => obj.id === props.participantId).name}</strong>gave a thumbs down 
        <i className="fas fa-thumbs-down"></i></p>
      </div>)
  }
  else if (props.type === 'raise-hand'){
    return (
      <div className ='raise-hand'>
        <p><strong>{STORE.participants.find(obj => obj.id === props.participantId).name}</strong>raised their hand 
        <i className="fas fa-hand-paper"></i></p>
      </div>)
  }
  else if (props.type === 'clap'){
    return (
      <div className ='clap'>
        <p><strong>{STORE.participants.find(obj => obj.id === props.participantId).name}</strong>clapped 
        <i className="fas fa-sign-language"></i></p>
      </div>)
  }
  else if (props.type === 'join'){
    return (
      <div className ='join'>
        <p><strong>{STORE.participants.find(obj => obj.id === props.participantId).name}</strong>joined</p>
      </div>)
  }
  else if (props.type === 'leave'){
    return (
      <div className ='leave'>
        <p><strong>{STORE.participants.find(obj => obj.id === props.participantId).name}</strong>left</p>
      </div>)
  }
  else if (props.type === 'join-stage'){
    return (
      <div className ='join-stage'>
        <p><strong>{STORE.participants.find(obj => obj.id === props.participantId).name}</strong>joined stage</p>
      </div>)
  }
  else if (props.type === 'leave-stage'){
    return (
      <div className ='leave-stage'>
        <p><strong>{STORE.participants.find(obj => obj.id === props.participantId).name}</strong>left stage</p>
      </div>)
  }
}
