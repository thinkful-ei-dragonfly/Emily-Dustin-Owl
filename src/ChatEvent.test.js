import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatEvent from './ChatEvent';



describe('Chat Event component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatEvent 
      key = {0} 
      participantId={2} 
      type={'message'}
      time={1548852544247}
      message={'Hello'}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expexted', () => {
    const tree = renderer
      .create(<ChatEvent 
        key={0} 
        participantId={2} 
        type={'message'}
        time={1548852544247}
        message={'Hello'}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

});