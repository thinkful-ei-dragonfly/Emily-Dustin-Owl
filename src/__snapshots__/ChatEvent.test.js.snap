// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Event component renders the UI as expexted 1`] = `
<div
  className="message"
>
  <img
    alt="user avatar"
    src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
  />
  <h3>
    Caty Flucker
  </h3>
  <p>
    5:49:04 AM
  </p>
  <p
    className="message-text"
  >
    Hello
  </p>
</div>
`;
